src/shared/proto/nanopb.bzl

```diff
@@ -65,6 +75,11 @@ def _nanopb_proto_library_impl(ctx):
         if label[CcInfo].linking_context != None
     ]
 
+    # The absolute path to the root folder for all generated files
```

Comment:
Just confirming, this is the base of the "duplicated folder structure" right? Would be really helpful to have a comment explaining this structure and why we need it here.

ie. bazel doesn't let you delcare / generate files outside the immediate folder/scope, so nanopb can't transitively generate nanopb files in other proto folders (like software)

Reply:
Great idea. Added the following above `generation_folder_name` above:
```
    # Because we need to generate the `.c` and `.h` files for all the proto files required
    # for this library, including all the transitive dependencies, we would like to
    # generate these in the same folder as each proto file. However bazel will not
    # permit us to modify/produce files outside the folder that this rule is called from.
    # We get around this be reproducing the folder structer under a generation folder
    # and adding the root of this directory as an include path.
```
then moved this variable up to under that and renamed it `generated_folder_abs_path`